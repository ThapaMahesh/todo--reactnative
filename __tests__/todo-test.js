import React from 'react';
import renderer from 'react-test-renderer';

import Todo from '../todo';

describe('<Todo />', () => {
  it('renders', () => {
    const tree = renderer.create(<Todo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('add-item', () => {
    const todoinst = renderer.create(<Todo />).getInstance();
    todoinst.addItemToList("new text");

    expect(todoinst.state.itemList).toEqual([{id: 1, title: "new text", isDone: false}]);
  });

  it('remove-item', () => {
    const todoinst = renderer.create(<Todo />).getInstance();
    todoinst.removeItem(1);

    expect(todoinst.state.itemList).toEqual([]);
  });
});