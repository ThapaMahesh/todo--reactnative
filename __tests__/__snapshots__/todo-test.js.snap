// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> renders 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "width": "100%",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          Todo List
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "width": "100%",
        },
      ]
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <RCTScrollView
        refreshControl={
          <RefreshControlMock
            onRefresh={[Function]}
            refreshing={false}
          />
        }
      >
        <RCTRefreshControl />
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "padding": 5,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              onChangeText={[Function]}
              placeholder="Add new item"
              placeholderTextColor="#575757"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 5,
                    "paddingRight": 5,
                  },
                  Object {
                    "borderBottomWidth": 1,
                    "borderStyle": "dotted",
                  },
                ]
              }
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
            />
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "marginLeft": 15,
                  "opacity": 1,
                  "padding": 15,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
            >
              <Text>
                Add
              </Text>
            </View>
          </View>
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            extraData={
              Object {
                "currentIndex": 0,
                "itemList": Array [],
                "refreshing": false,
              }
            }
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
